core/src/pretty.rs

```diff
@@ -152,17 +153,197 @@ fn needs_parens_in_type_pos(typ: &Type) -> bool {
     }
 }
 
-pub fn fmt_pretty<'a, T>(value: &T, f: &mut fmt::Formatter) -> fmt::Result
+pub fn fmt_pretty<T>(value: &T, f: &mut fmt::Formatter) -> fmt::Result
 where
-    T: Pretty<'a, pretty::BoxAllocator, ()> + Clone,
+    T: for<'a> Pretty<'a, BoundedAllocator, ()> + Clone,
```

Comment:
Out of curiosity, what is the reason to move the `forall 'a` into `T` instead of `fmt_pretty`?